plugins/DeepJetTableProducer.cc

```diff
@@ -0,0 +1,406 @@
+            Cpfcan_quality_0to24[c_pf_n][i_jet] = c_pf_features.quality;
+          }
+          
+          // n_pf candidates
+          auto max_n_pf_n = std::min(features.n_pf_features.size(), (std::size_t)25);
+          for (std::size_t n_pf_n = 0; n_pf_n < max_n_pf_n; n_pf_n++) {
+            const auto& n_pf_features = features.n_pf_features.at(n_pf_n);
+            Npfcan_ptrel_0to24[n_pf_n][i_jet] = n_pf_features.ptrel;
+            Npfcan_deltaR_0to24[n_pf_n][i_jet] = n_pf_features.deltaR;
+            Npfcan_isGamma_0to24[n_pf_n][i_jet] = n_pf_features.isGamma;
+            Npfcan_HadFrac_0to24[n_pf_n][i_jet] = n_pf_features.hadFrac;
+            Npfcan_drminsv_0to24[n_pf_n][i_jet] = n_pf_features.drminsv;
+            Npfcan_puppiw_0to24[n_pf_n][i_jet] = n_pf_features.puppiw;
+          }
+
+          // sv candidates
+          auto max_sv_n = std::min(features.sv_features.size(), (std::size_t)4);
+          for (std::size_t sv_n = 0; sv_n < max_sv_n; sv_n++) {
+            const auto& sv_features = features.sv_features.at(sv_n);
+            sv_pt_0to3[sv_n][i_jet] = sv_features.pt;
+            sv_deltaR_0to3[sv_n][i_jet] = sv_features.deltaR;
+            sv_mass_0to3[sv_n][i_jet] = sv_features.mass;
+            sv_ntracks_0to3[sv_n][i_jet] = sv_features.ntracks;
+            sv_chi2_0to3[sv_n][i_jet] = sv_features.chi2;
+            sv_normchi2_0to3[sv_n][i_jet] = sv_features.normchi2;
+            sv_dxy_0to3[sv_n][i_jet] = sv_features.dxy;
+            sv_dxysig_0to3[sv_n][i_jet] = sv_features.dxysig;
+            sv_d3d_0to3[sv_n][i_jet] = sv_features.d3d;
+            sv_d3dsig_0to3[sv_n][i_jet] = sv_features.d3dsig;
+            sv_costhetasvpv_0to3[sv_n][i_jet] = sv_features.costhetasvpv;
+            sv_enratio_0to3[sv_n][i_jet] = sv_features.enratio;
+          }
+      }
+  }
+    
+  // DeepJetInputs table
+  auto djTable = std::make_unique<nanoaod::FlatTable>(jet_N_CPFCands.size(), nameDeepJet_, false, true);
+  //djTable->addColumn<int>("DeepJet_jetIdx", jetIdx_dj, "Index of the parent jet", nanoaod::FlatTable::IntColumn);
+    
+    
+  djTable->addColumn<int>("DeepJet_nCpfcand", jet_N_CPFCands, "Number of charged PF candidates in the jet", nanoaod::FlatTable::IntColumn);
+  djTable->addColumn<int>("DeepJet_nNpfcand", jet_N_NPFCands, "Number of neutral PF candidates in the jet", nanoaod::FlatTable::IntColumn);
+  djTable->addColumn<int>("DeepJet_nsv", jet_N_SVs, "Number of secondary vertices in the jet", nanoaod::FlatTable::IntColumn);
+  djTable->addColumn<int>("DeepJet_npv", jet_N_PVs, "Number of primary vertices", nanoaod::FlatTable::IntColumn);
+    
+    
+  std::string input_name;
+  std::string description;  
+  for (unsigned int p = 0; p < 25; p++) {
+      auto s = std::to_string(p);
+      
+      
+      // ============================================================== Cpfs ===================================================================
+      input_name = "DeepJet_Cpfcan_puppiw_" + s;
```

Comment:
Maybe these can be formated better? Defining and overwriting `description` and `inpu_name` for each entry is redundant